components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineSession.kt

```diff
@@ -338,13 +348,18 @@ class SystemEngineSession(
 
     /**
      * See [EngineSession.toggleDesktopMode]
+     *
+     * Precondition:
+     * If settings.useWideViewPort = true, then desktopMode is always on (will never be toggled)
+     * If settings.useWideViewPort = false or null, then desktopMode can be toggled on/off
```

Comment:
Love the idea to add a comment here explaining side effects. But isn't the comment a bit misleading? You can still continue to use toggleDesktopMode() regardless of what you've set useWideViewPort to?

Reply:
Ah okay, I misunderstood the toggle logic